doc.go

```diff
@@ -29,6 +29,7 @@ type Document struct {
 	fieldNames      []string
 	fieldTokenFreqs []index.TokenFrequencies
 	fieldLens       []int
+	vectorDims      []int // applicable to vector fields only
```

Comment:
i didn't understand why the datatype for the vectorDims is an []int instead of int. is it for like per field?

Reply:
Because there can be multiple vector fields over which multi knn can be run :)

Reply:
correct, we don't allow vector fields with dims = 0 anyway - so should be good there.